### PR TITLE
Network backend updates

### PR DESCRIPTION
**Pull request by bloemp**
Hi @matmair,
I've updated the network backend to allow networked printers be discoverable through SNMP and report back its status.
The reason for starting this PR is so we can modify the inventree-brother plugin to at least use auto-detect for its media type (changing it in the settings for different type of labels is cumbersome). And maybe even use the auto-discover for easy access (but SNMP broadcasts from a docker container needs --network=host).
I know that this a fork of the original, but as the original is basically unmaintained, I thought I would directly PR it to you.

### CODE REVIEW

File: brother_ql/backends/helpers.py [hunk 1]
```diff
@@ -63,9 +64,6 @@ def send(instructions, printer_identifier=None, backend_identifier=None, blockin
 
     if not blocking:
         return status
-    if selected_backend == 'network':
-        """ No need to wait for completion. The network backend doesn't support readback. """
-        return status
```
Comment (matmair) [inline brother_ql/backends/helpers.py:68]: can you expand on this change?

File: brother_ql/backends/network.py [hunk 2]
```diff
@@ -8,10 +8,38 @@
...
 
+# Some SNMP OID's that we can use to get printer information
+brother_ql_snmp_oids = {
+    "snmp_get_ip"       : "1.3.6.1.4.1.1240.2.3.4.5.2.3.0",
+    "snmp_get_netmask"  : "1.3.6.1.4.1.1240.2.3.4.5.2.4.0",
+    "snmp_get_mac"      : "1.3.6.1.4.1.1240.2.3.4.5.2.4.0",
+    "snmp_get_location" : "1.3.6.1.2.1.1.6.0",
+    "snmp_get_model"    : "1.3.6.1.2.1.25.3.2.1.3.1",
+    "snmp_get_serial"   : "1.3.6.1.2.1.43.5.1.1.17",
+    "snmp_get_status"   : "1.3.6.1.4.1.2435.3.3.9.1.6.1.0"
```
Comment (matmair) [inline brother_ql/backends/network.py:34]: Question: Wouldn't enums or a class be a safer solution here (thinking about misstyped references)

File: brother_ql/cli.py [hunk 3]
```diff
@@ -120,7 +120,7 @@ def env(ctx, *args, **kwargs):
     print("\n##################\n")
 
 @cli.command('print', short_help='Print a label')
-@click.argument('images', nargs=-1, type=click.File('rb'), metavar='IMAGE [IMAGE] ...')
+@click.argument('images', nargs=-1, required=True, type=click.File('rb'), metavar='IMAGE [IMAGE] ...')
```
Comment (matmair) [inline brother_ql/cli.py:123]: Question: Is this directly related to network printing? If so how?

File: brother_ql/reader.py [hunk 4]
```diff
@@ -66,9 +66,12 @@
 
 RESP_MEDIA_TYPES = {
   0x00: 'No media',
-  0x01: 'Laminated Tape',
+  0x01: 'Laminated tape',
+  0x03: 'Non-laminated tape',
   0x0A: 'Continuous length tape',
   0x0B: 'Die-cut labels',
+  0x11: 'Heat-shrink tube',
+  0xff: 'Incompatible tape',
```
Comment (matmair) [inline brother_ql/reader.py:74]: Question: Is this directly related to network printing? If so how?

**Comment by matmair** [comment]
@bloemp I have a few questions.
In general, I am a bit hesitant about this change. I do not have access to a network-based printer currently so it is hard for me to gauge how/if this breaks anything.
Would you be able to also open a PR against upstream? I know the maintainer is not very responsive but if that changes I would like to phase out this fork ASAP.

File: brother_ql/backends/helpers.py (hunk 1 above)
Comment (bloemp) [reply brother_ql/backends/helpers.py:68]: These lines are removed, because the network backend now does support readback. 
Therefore we can continue executing the function and fill in and return the actual status values.

File: brother_ql/backends/network.py (hunk 2 above)
Comment (bloemp) [reply brother_ql/backends/network.py:34]: Thats a good idea, will change the code accordingly

File: brother_ql/reader.py (hunk 4 above)
Comment (bloemp) [reply brother_ql/reader.py:74]: This isn't directly related to network printing... while I like the structure to be complete, it has no real place in this PR. Will revert that change also

**Comment by bloemp** [comment]
@matmair I've updated the PR with your input (which is very much appreciated).
I understand why you are hesitant to merge this as you can't test it yourself and would rather see this fork disappear asap.
Therefore, I think it is best to close this PR and create one against the original repo. Hopefully, someone can test is there, and the original maintainer will merge in the future.

**Comment by matmair** [comment]
@bloemp if we find a second user that can verify I am happy to merge, I appreciate the work you put into discovery.
I will put a note in the next InvenTree newsletter (should go out at the end of January).

**Comment by matmair** [comment]
It is more about having multiple environments then thinking the code is wrong. Hardware interfaces are tricky.
